packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.styles.ts

```diff
@@ -339,7 +339,8 @@ export const getStyles = (props: IDetailsHeaderStyleProps): IDetailsHeaderStyles
     dropHintStyle: [
       {
         display: 'inline-block',
-        position: 'absolute'
+        position: 'absolute',
+        overflow: 'hidden'
```

Comment:
Can you quickly explain why there are 2 different overflow changes in this PR? This one in styles, and the other in resetDropHints. Are those the same css values? Or two different elements?

Reply:
I was taking the wrong approach when I submitted the PR. The issue seems to be resolved when I just change `.ms-DetailsList` to `overflow: visible`. That class already has `overflow-y: visible`, so I was wondering why it had `overflow-x: auto`.

The `DetailsList` component doesn't seem to support x-scrolling, even when it's so narrow that some columns aren't visible. So I don't see the harm in making `.ms-DetailsList` have `overflow: visible`.

If you agree with the the `overflow: visible` solution, I can submit another PR.

Reply:
But making `overflow-x: visible` would mean even when content starts overflowing horizontally, we wouldn’t have a scroll bar.

![image](https://user-images.githubusercontent.com/15196876/47707231-52896d80-dc51-11e8-9a82-c1ba384a7b79.png)

Reply:
The scrollbar doesn't do anything right now. When content overflows horizontally, columns are hidden and/or cell contents get ellipses. Horizontal scrolling doesn't do anything besides create an unnecessary horizontal scrollbar.

Reply:
There is a mode in DetailsList to allow for horizontal scrolling. See the `constrainMode` prop.

Reply:
I took a look at `constrainMode`. It defaults to `horizontalConstrained`, which is setting `overflowX: 'auto'`. Is the purpose of this mode to allow horizontal scrolling when there isn't enough width to display all columns? If so, then it isn't working as expected. The issue raised by this PR is caused by `horizontalConstrained`, and I'm not seeing any benefit from `horizontalConstrained`.

Setting `constrainMode={unconstrained}` solves the issue raised by this PR.

Please let me know what `constrainMode={horizontalConstrained}` is supposed to do.

Reply:
Judging from the _only_ usage of that prop, it seems like the intent of `horizontalConstrained` is to ensure that the component overflows in the `X` direction by setting the according styles:

https://github.com/OfficeDev/office-ui-fabric-react/blob/ea8aacd189d53a164e95c3535c3170e447314ec2/packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.styles.ts#L49

I don't have full context as the prop predates my involvement with OUFR, but that is my reading of it.

Reply:
Yea, I agree that seems like the purpose, but does it actually accomplish that? The only effect it seems to have is creating a scrollbar that moves a few pixels (i.e. the issue which prompted this PR).

Reply:
I think there's still a valid use-case for having a prop which constrains width and therefore requires `overflow-x: auto`. I've isolated the root cause of the above linked issue. Let me solve that and then perhaps I can have time to demonstrate the `isHorizontalConstrained` prop's usefulness. If it isn't useful then we can deprecate.